pkg/reconciler/iploop.go

```diff
@@ -90,23 +92,73 @@ func (rl *ReconcileLooper) findOrphanedIPsPerPool(ipPools []storage.IPPool) erro
 func (rl ReconcileLooper) isPodAlive(podRef string, ip string) bool {
 	for livePodRef, livePod := range rl.liveWhereaboutsPods {
 		if podRef == livePodRef {
-			if livePod.phase == v1.PodPending {
-				logging.Debugf("PF9: Skipping Pod IP check for Pending Pod: %s IP: %s", livePodRef, ip)
-				return true
+			isFound := isIpOnPod(&livePod, podRef, ip)
+			if !isFound && (livePod.phase == v1.PodPending) {
+				/* Sometimes pods are still coming up, and may not yet have Multus
+				 * annotation added to it yet. We don't want to check the IPs yet
+				 * so re-fetch the Pod 5x
+				 */
+				podToMatch := &livePod
+				retries := 0
+
+				logging.Debugf("Re-fetching Pending Pod: %s IP-to-match: %s", livePodRef, ip)
+
+				for retries < storage.PodRefreshRetries {
+					retries += 1
```

Comment:
Minor NIT - but maybe also log the retries

Reply:
It is already logged, in the refreshPod function, for each retry we will see:

logging.Debugf("Got refreshed pod: %v", wrappedPod)

No need to log the retries, since its evident based on each refresh Pod log